test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/AttributeUtilityTests.cs

```diff
@@ -40,108 +50,197 @@ public void CreateCommitmentTypeIndication_WithUnknownSignature_Throws()
         }
 
         [Fact]
-        public void GetCommitmentTypeIndication_WithBothOriginAndReceiptTypes_Throws()
+        public void GetSignatureType_WithBothOriginAndReceiptTypes_Throws()
         {
             var attribute = GetCommitmentTypeTestAttribute(
                 Oids.CommitmentTypeIdentifierProofOfOrigin, Oids.CommitmentTypeIdentifierProofOfReceipt);
+            var attributes = new CryptographicAttributeObjectCollection(attribute);
 
             var exception = Assert.Throws<SignatureException>(
-                () => AttributeUtility.GetCommitmentTypeIndication(attribute));
+                () => AttributeUtility.GetSignatureType(attributes));
 
             Assert.Equal("The commitment-type-indication attribute contains an invalid combination of values.", exception.Message);
         }
 
         [Fact]
-        public void GetCommitmentTypeIndication_WithDuplicateOriginTypes_ReturnsAuthor()
+        public void GetSignatureType_WithDuplicateOriginTypes_ReturnsAuthor()
         {
             var attribute = GetCommitmentTypeTestAttribute(
                 Oids.CommitmentTypeIdentifierProofOfOrigin, Oids.CommitmentTypeIdentifierProofOfOrigin);
+            var attributes = new CryptographicAttributeObjectCollection(attribute);
 
-            Assert.Equal(SignatureType.Author, AttributeUtility.GetCommitmentTypeIndication(attribute));
+            Assert.Equal(SignatureType.Author, AttributeUtility.GetSignatureType(attributes));
         }
 
         [Fact]
-        public void GetCommitmentTypeIndication_WithOriginAndUnknownType_ReturnsAuthor()
+        public void GetSignatureType_WithOriginAndUnknownType_ReturnsAuthor()
         {
             var attribute = GetCommitmentTypeTestAttribute(
                 Oids.CommitmentTypeIdentifierProofOfOrigin, CommitmentTypeIdentifierProofOfSender);
+            var attributes = new CryptographicAttributeObjectCollection(attribute);
 
-            Assert.Equal(SignatureType.Author, AttributeUtility.GetCommitmentTypeIndication(attribute));
+            Assert.Equal(SignatureType.Author, AttributeUtility.GetSignatureType(attributes));
         }
 
         [Fact]
-        public void GetCommitmentTypeIndication_CommitmentTypeIndicationWithBothRepoAndAuthorVerifyThrows()
+        public void GetSignatureType_WithOriginTypeInNonFirstAttributeInstance_ReturnsAuthor()
+        {
+            using (var certificate = _fixture.GetDefaultCertificate())
+            {
+                var attributes = CreateAttributeCollection(certificate, _fixture.DefaultKeyPair.Private,
+                    vector =>
+                    {
+                        vector.Add(
+                            new Org.BouncyCastle.Asn1.Cms.Attribute(
+                                PkcsObjectIdentifiers.IdAAEtsCommitmentType,
+                                new DerSet(
+                                    new BcCommitmentTypeIndication(PkcsObjectIdentifiers.IdCtiEtsProofOfApproval))));
```

Comment:
🏆 glad you were able to get this working! I had so much trouble creating the bouncy castle version of our attributes.